python/cudf/cudf/bindings/stream_compaction.pyx

```diff
@@ -70,24 +73,42 @@ def apply_drop_duplicates(in_index, in_cols, subset=None, keep='first'):
     return (out_cols[:-1], out_cols[-1])
 
 
-def cpp_apply_boolean_mask(inp, mask):
-    from cudf.dataframe.columnops import column_empty_like
+def apply_apply_boolean_mask(cols, mask):
+
+    cdef cudf_table  c_out_table
+    cdef cudf_table* c_in_table = table_from_columns(cols)
+    cdef gdf_column* c_mask_col = column_view_from_column(mask)
 
-    cdef gdf_column *inp_col = column_view_from_column(inp)
-    cdef gdf_column *mask_col = column_view_from_column(mask)
-    cdef gdf_column result
     with nogil:
-        result = apply_boolean_mask(inp_col[0], mask_col[0])
-    if result.data is NULL:
-        return column_empty_like(inp, newsize=0)
-    data, mask = gdf_column_to_column_mem(&result)
-    return Column.from_mem_views(data, mask)
+        c_out_table = apply_boolean_mask(c_in_table[0], c_mask_col[0])
+
+    free(c_in_table)
+    free(c_mask_col)
+
+    return columns_from_table(&c_out_table)
+
 
+def apply_drop_nulls(cols, how="any", subset=None, thresh=None):
```

Comment:
Need an assertion that `subset` is a non-empty list if it is not `None`

Reply:
If it's empty, cudf::drop_nulls will return an empty table like the input, so I don't think an assertion is required. Accepting empty tables and columns is a requirement for dask, I've been told multiple times.

Reply:
@shwina Please do a "request changes" review if you still think this is needed so this doesn't get merged.

Reply:
Actually, if subset is empty, `apply_drop_nulls` should just return `cols` unchanged. This is being handled in the `dropna()` PR so I'll mark as resolved